scripts/jenkins/ardana/gerrit/gerrit_handle_event.py

```diff
@@ -10,6 +12,38 @@
 from gerrit_review import gerrit_review  # noqa: E402
 
 
+def jenkins_build_job(job_name, job_args=[]):
```

Comment:
I know this makes things a bit more difficult, but I still believe it is worth keeping Jenkins away from these Gerrit scripts. For one, it makes the implementation more modular, or at least not look too much like spaghetti, and that's always a good thing to have.

In fact, the anomalous hard-coded job names below almost seem to suggest that this approach deviates from the trend that was set so far: that of having Jenkins as the top-level coordinator of tasks and information and keeping everything else as Jenkins agnostic as possible.

I propose to you that implementing this the other way around will "feel" better. This is what I mean by that: Jenkins calls the `gerrit_handle_event` script to 1. post the "New patchset was published for..." Gerrit message and 2. return a list of reference changes. Then, it calls the existing `scripts/jenkins/jenkins-job-trigger` script to basically trigger the Gerrit Jenkins job for each returned reference change. The job name doesn't need to be hardcoded anywhere, because it just so happens to be the same job that is actually running all this.

Reply:
Yep, this makes sense. At the moment I've moved the logic back into openstack-ardana-gerrit-events. I'll look at templating it as a followup commit.